ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DirectionsQuestionnaireMapper.java

```diff
@@ -0,0 +1,77 @@
+package uk.gov.hmcts.cmc.ccd.mapper;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.ccd.domain.CCDYesNoOption;
+import uk.gov.hmcts.cmc.ccd.domain.directionsquestionnaire.CCDDirectionsQuestionnaire;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.DirectionsQuestionnaire;
+import uk.gov.hmcts.cmc.domain.models.response.YesNoOption;
+
+import java.util.stream.Collectors;
+
+@Component
+public class DirectionsQuestionnaireMapper implements Mapper<CCDDirectionsQuestionnaire, DirectionsQuestionnaire> {
+
+    private final ExpertRowMapper expertRowMapper;
+    private final UnavailableDateMapper unavailableDateMapper;
+
+    @Autowired
+    public DirectionsQuestionnaireMapper(ExpertRowMapper expertRowMapper, UnavailableDateMapper unavailableDateMapper) {
+        this.expertRowMapper = expertRowMapper;
+        this.unavailableDateMapper = unavailableDateMapper;
+    }
+
+    @Override
+    public CCDDirectionsQuestionnaire to(DirectionsQuestionnaire directionsQuestionnaire) {
+        return CCDDirectionsQuestionnaire.builder()
+            .selfWitness(CCDYesNoOption.valueOf(directionsQuestionnaire.isSelfWitness()))
+            .howManyOtherWitness(directionsQuestionnaire.getHowManyOtherWitness())
+            .hearingLocation(directionsQuestionnaire.getHearingLocation())
+            .hearingLocationSlug(directionsQuestionnaire.getHearingLocationSlug())
+            .exceptionalCircumstancesReason(directionsQuestionnaire.getExceptionalCircumstancesReason())
+            .unavailableDates(directionsQuestionnaire.getUnavailableDates()
+                .stream()
```

Comment:
just FYI unavailableDates and exportReportRows can be null because of lombok + jackson sticking null instead of empty list in the directions questionnaire. I've solved it in another piece of code like this https://github.com/hmcts/cmc-claim-store/blob/master/ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/legaladvisor/CCDOrderGenerationData.java but there are other ways to skin the cat. Just something to keep in mind to avoid a NPE

Reply:
:)

Reply:
Please use `asStream` utility method which is null safe